syntax = "proto3";
package Proto;

import "Exchange.proto";
import "Request.proto";
import "Reply.proto";

enum PricingModel
{
  BSM = 0;
  CRR = 1;
}

enum UnderlyingTheoType
{
  Midpoint = 0;
  BaryCentre = 1;
  DepthBary = 2;
}

message Pricer
{
  string name = 1;
  string underlying = 2;
  PricingModel model = 3;
  repeated string options = 4;
  int32 depth = 5;
  int32 interval = 6;
  UnderlyingTheoType theo_type = 7;
  int32 warn_tick_change = 8;
  double elastic = 9;
  double elastic_limit = 10;
}

message PricerReq
{
  RequestType type = 1;
  Exchange exchange = 2;
  repeated Pricer pricers = 3;
  string name = 4;
  string user = 5;
}

message PricerRep
{
  repeated Pricer pricers = 1;
  Reply result = 2;
}
